docs/chapters/tutorials/basic-input.rst

```diff
@@ -3,45 +3,363 @@
 
 Modify the "Hello World!" App
 -----------------------------
 
+Clone the previous "Hello World!" app from the Git repo (be sure to replace Git repo URL with the appropriate URL for your app):
+
+.. code-block:: text
+
+    cd ~/geneflow_work
+    git clone https://github.com/[USER]/hello-world-gf.git
+
+Note: if the ``hello-world-gf`` directory still exists with the contents of the previous app tutorial, do a ``git pull`` instead:
+
+.. code-block:: text
+
+    cd ~/geneflow_work/hello-world-gf
+    git pull
+
 Define an App Input
 ~~~~~~~~~~~~~~~~~~~
 
-Re-"Make" the App
-~~~~~~~~~~~~~~~~~
+Update the app metadata to reflect the new functionality:
+
+.. code-block:: text
+
+    vi ./config.yaml
+
+Change the ``description`` field to indicate a change in the app; add a ``repo_uri`` field (with [USER] replaced with your username, and github.com replaced with the appropriate Git service); update the ``version`` field to ``0.2``:
+
+.. code-block:: yaml
+
+    # Basic app information
+
+    # name: standard GeneFlow app name
+    name: hello-world-gf
+    # description: short description for the app
+    description: Updated hello world GeneFlow app
+    # repo_uri: link to the app's git repo
+    repo_uri: https://github.com/[USER]/hello-world-gf.git
+    # version: must be incremented every time this file, or any file in the app
+    # project is modified
+    version: '0.2'
+
+Next, update the ``file`` input in the ``inputs`` section of the ``config.yaml`` file. Change the ``label`` and ``description`` of the ``file`` input to ``Input text file``. Change the ``required`` field to ``true`` and add a ``test_value`` field with value ``${SCRIPT_DIR}/data/file.txt``. The ``test_value`` field is only used when running the app's test script. We will define the ``/data/file.txt`` file in the next section. ``${SCRIPT_DIR}`` in the context of the test script is the directory in which the test script is located. The updated ``inputs`` section of the ``config.yaml`` file should look like:
+
+.. code-block:: yaml
+
+    inputs:
+      file:
+        label: Input Text File
+        description: Input text file
+        type: File
+        required: true
+        test_value: ${SCRIPT_DIR}/data/file.txt
+
+Finally, change the execution commands of the app in the ``exec_methods`` section. 
+
+.. code-block:: yaml
+
+    exec_methods:
+    - name: environment
+      if:
+      - in_path: 'cat'
+      exec:
+      - run: cat ${FILE_FULL}
+        stdout: ${OUTPUT_FULL}
+
+The modified execution block first checks if the ``cat`` command is available (it should be available in all standard Linux systems). It then runs the ``cat`` command to print the contents of the file passed as the ``file`` input. ``${FILE_FULL}`` is a bash variable that is automatically defined for the ``file`` input, and represents the full path of the ``file`` input. 
```

Comment:
Do you think we should change the capitalization for the word "bash" to "Bash"?

Reply:
Fixed in the GF-405-basic-inputs branch.